### ISSUE TITLE
Trouble with XYChart.removeData

### ISSUE THREAD

**Issue by th3ragex** [amcharts/amcharts4#1816]
I am a bit unsure whether the XYChart.removeData method is currently working as expected.

For testing i have created a fiddle based on the live data demo (https://www.amcharts.com/demos/live-data/)
Fiddle: http://jsfiddle.net/0ymso2pb/31/

The expected behavior: The oldest values gets removed periodically
Current behavior: Internal error "m is undefined"
Workaround: chart.addData(null, 1); or chart.addData([], 1);

In this code part please try out including chart.addData(null, 1); and removing    chart.removeData(1); to get the expected behavior:
``` javascript
var interval;
function startInterval() {
    interval = setInterval(function() {
        visits =
            visits + Math.round((Math.random() < 0.5 ? 1 : -1) * Math.random() * 5);
        var lastdataItem = series.dataItems.getIndex(series.dataItems.length - 1);
        //chart.addData({ date: new Date(lastdataItem.dateX.getTime() + 1000), value: visits });
        chart.removeData(1);  <--------------
        //chart.addData(null, 1);  <--------------
    }, 1000);
}
```

This problem ocurres in my bigger line chart too and i am a bit clueless on how to fix this. The removeData() call directly after an addData works for some reason.

**Comment by martynasma** [comment]
Yes, there's something wonky with standalone usage of `removeData()`, indeed.

We'll be looking into it.

**Comment by martynasma** [comment]
Fixed in [4.7.7](https://github.com/amcharts/amcharts4/releases/tag/4.7.7).

## [4.7.7] - 2019-11-02

### Added
- `NumberFormatter` now supports `numberFormat` in [Intl.DateTimeFormatOptions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat) objects. E.g. `chart.numberFormatter.numberFormat = { style: "decimal", minimumFractionDigits: 3 }`. [More info](https://www.amcharts.com/docs/v4/tutorials/formatting-date-time-and-numbers-using-intl-object/).
- New built-in pattern: `CirclePattern`.
- `PictorialStackedSeries` now supports `align`/`valign` settings, allowing controlling its position inside chart container.

### Changed
- Default pattern list in `PatternSet` is expanded and revamped for better look and increased contrast.
- Pattern elements are now centered, so they work better with `rotation`.
- `MapChart` now will shift zoom towards center gradually when zooming out.

### Fixed
- `getIndex()` method of `PatternSet` was always returning current pattern, not particular index.
- Sometimes Export's user-added `"data"` adapter was kicking in before amCharts built-in adapter, causing data edits to be overwritten.
- Built-in adapters (e.g. for export `"data"`) sometimes were being added multiple times.
- Export no longer errors out on pages with missing external CSS files.
- Using `addData()` on `PercentChart` was reusing colors from the beginning for the newly added slices.
- `SliceGrouped` on `PercentSeries` when used on a chart with responsive fetures enabled, could display labels of the hidden slices after chart resize.
- Issue with zooming `DateAxis` with single data item fixed (the single item used to disappear in certain cases).

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
